datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -67,6 +68,8 @@ public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<In
 
     LOG.info("old partition assignment info, assignment: {}", currentAssignment);
 
+    Validate.isTrue(currentAssignment.size() > 0, "Zero tasks assigned. Retry leader assignment.");
```

Comment:
just curious, do we know why currentAssignment size was 0? I know saw this happen once on ZK session expiry and looked like the ZKClient was null (had the fix Justin made)

Reply:
Also the log message, can we change it to: "Retry leader partition assignment"?

Reply:
Unfortunately, I don't have the logs to confirm why. I thought it is still a good fix irrespective of the cause.